src/util/pagination.js

```diff
@@ -51,9 +81,177 @@ async function offsetPaginate ({
   return paginationMeta
 }
 
+// ///////////////// //
+// CURSOR PAGINATION //
+// ///////////////// //
+
+/**
+ * @param {Object}   queryBuilder - Knex.js query builder
+ *
+ * `startingAfter` and `endingBefore` are mutually exclusive
+ * @param {String}   [startingAfter] - if specified, fetch results after this cursor
+ * @param {String}   [endingBefore] - if specified, fetch results before this cursor
+ *
+ * @param {Number}   nbResultsPerPage
+ *
+ * @param {Object[]} cursorConfig - will be used to create/parse cursor
+ * @param {Object}   cursorConfig[i]
+ * @param {String}   cursorConfig[i].prop - object property to encode
+ * @param {String}   cursorConfig[i].type - allowed value: 'number', 'boolean', 'date', 'string'
+ *
+ * @param {String}   order - allowed values: 'asc', 'desc'
+ */
+async function cursorPaginate ({
+  queryBuilder,
+  startingAfter,
+  endingBefore,
+  nbResultsPerPage,
+  cursorConfig,
+  order,
+}) {
+  // if `endingBefore` is specified, this is equivalent to retrieving
+  // the `nbResultsPerPage` last results.
+  // To achieve it, this is done with 2 steps:
+  // 1. reverse the `order` direction for the SQL query with limit to `nbResultsPerPage`
+  // 2. reverse the obtained results in Javascript
+  const shouldReverseOrder = !!endingBefore
+
+  const cursor = startingAfter || endingBefore
```

Comment:
As in previous PRs:

Don’t you think we should only allow one of these? For least surprise, rather than silently picking one.

By the way, in a subsequent commit you make it exclusive https://github.com/stelace/stelace/pull/371/commits/044fabbbdefd1995afdc9c657d67d47de8cbc7a3#diff-285f19a9ab4a0e60e77a40695003d5ebR50

Reply:
I didn't make it exclusive in this util function because I relied on the validation step.
Do you think it will be safer if I also add a check here?